tests/plugins/find_and_replace.test.ts

```diff
@@ -290,163 +430,343 @@ describe("basic search", () => {
     deleteFilter(model, "A1:A6");
     matches = model.getters.getSearchMatches();
     matchIndex = model.getters.getCurrentSelectedMatchIndex();
-    expect(matches).toHaveLength(4);
+    expect(matches).toHaveLength(3);
     expect(matchIndex).toStrictEqual(0);
   });
 
   test("Need to maintain search if column or row is added", () => {
     model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
     let matches = model.getters.getSearchMatches();
     let matchIndex = model.getters.getCurrentSelectedMatchIndex();
-    expect(matches).toHaveLength(4);
+    expect(matches).toHaveLength(3);
     expect(matchIndex).toStrictEqual(0);
-    expect(matches[0]).toStrictEqual({ col: 0, row: 1, selected: true });
-    expect(matches[1]).toStrictEqual({ col: 0, row: 2, selected: false });
-    expect(matches[2]).toStrictEqual({ col: 0, row: 3, selected: false });
-    expect(matches[3]).toStrictEqual({ col: 0, row: 4, selected: false });
+    expect(matches[0]).toStrictEqual({ sheetId: "s1", col: 0, row: 1, selected: true });
+    expect(matches[1]).toStrictEqual({ sheetId: "s1", col: 0, row: 2, selected: false });
+    expect(matches[2]).toStrictEqual({ sheetId: "s1", col: 0, row: 3, selected: false });
     addRows(model, "after", 1, 1);
     matches = model.getters.getSearchMatches();
     matchIndex = model.getters.getCurrentSelectedMatchIndex();
-    expect(matches).toHaveLength(4);
+    expect(matches).toHaveLength(3);
     expect(matchIndex).toStrictEqual(0);
-    expect(matches[0]).toStrictEqual({ col: 0, row: 1, selected: true });
-    expect(matches[1]).toStrictEqual({ col: 0, row: 3, selected: false });
-    expect(matches[2]).toStrictEqual({ col: 0, row: 4, selected: false });
-    expect(matches[3]).toStrictEqual({ col: 0, row: 5, selected: false });
+    expect(matches[0]).toStrictEqual({ sheetId: "s1", col: 0, row: 1, selected: true });
+    expect(matches[1]).toStrictEqual({ sheetId: "s1", col: 0, row: 3, selected: false });
+    expect(matches[2]).toStrictEqual({ sheetId: "s1", col: 0, row: 4, selected: false });
   });
 });
+
 describe("next/previous cycle", () => {
```

Comment:
Yeah I didn't read all that because I kinda gave up, but I'm sure there no need for three whole hundreds and twenty lines of code to test "previous/next". And even less so after we tested that all the matches are correct for 450 lines in the previous tests